bindings/rust/wasmedge-sys/src/vm.rs

```diff
@@ -0,0 +1,174 @@
+use super::wasmedge;
+use crate::{
+    raw_result::{decode_result, ErrReport},
+    string::StringRef,
+    value::Value,
+    wasi,
+};
+use std::os::raw::c_char;
+
+// If there is a third-party sdk based on wasmedge-sys
+// the private encapsulation  here can force the third-party sdk to use the api
+#[derive(Debug)]
+pub struct Vm {
+    pub(crate) ctx: *mut wasmedge::WasmEdge_VMContext,
+    _private: (),
+}
+
+impl Vm {
+    pub fn create(config: &crate::config::Config) -> Result<Self, ErrReport> {
+        let ctx = unsafe {
+            wasmedge::WasmEdge_VMCreate(config.ctx, std::ptr::null_mut() /* store */)
+        };
+        if ctx.is_null() {
+            panic!("WasmEdge VM create failed");
```

Comment:
`panic` should only be used for irrecoverable errors. Here, it's likely because the C lib doesn't like the config, but that can be fixed by application code. If the only failure mode is because of OOM or something like that, you can disregard this comment.

For libraries, panicking is incredibly difficult to work around when it's not easy to check why the operation would fail (compare to `&str/&[T]` methods that panic when an index is wrong, but that's easy to check).

Reply:
The panic here is the irrecoverable error.